File: rachinha_front/interface/booking.ts
export interface Booking {
  price_type: "per_person" | "total_split" | undefined;
  price?: number | null;
  _id: string;
  court_id: string;
  start_time: string; // ou Date
  end_time: string; // ou Date
  modality: string;
  status: string;
  created_at: string; // ou Date
  updated_at: string; // ou Date
  max_players: number;
  players: Player[];
  reserve_players: Player[];
  owner_id: string;
  location: {
    alt: string;
  }
  associated_group_id?: string;
  status_list: boolean;
  organized_teams?: BookingOrganizedTeams | null;
}

export interface BookingOrganizedTeams {
  teams?: string[][];
  team_skills_sum?: number[];
  reserves?: string[];
  drawn_at?: string;
  drawn_by_name?: string;
  result?: {
    teams: string[][];
    team_skills_sum: number[];
    reserves?: string[];
  } | null;
}

export interface Player {
  _id: string;
  user_id: string;
  skill_level: number;
  status: string;
}

export type BookingUpdate = Partial<Booking>;

File: rachinha_back/app/interfaces/schemas/booking.py
from pydantic import BaseModel
from datetime import datetime
from typing import Literal, Optional, List

from app.interfaces.schemas.group import Player

class BookingUpdate(BaseModel):
    status: Optional[str] = None
    start_time: Optional[datetime] = None
    end_time: Optional[datetime] = None
    modality: Optional[str] = None
    max_players: Optional[int] = None
    players: Optional[List[Player]] = None
    reserve_players: Optional[List[Player]] = None
    associated_group_id: Optional[str] = None 
    location: Optional[dict] = None
    status_list: Optional[bool] = None
    price: Optional[float] = None
    price_type: Optional[Literal["per_person", "total_split"]] = None


class RecurringBookingCreate(BaseModel):
    start_time: datetime
    end_time: datetime
    modality: str
    max_players: int = 0
    players: Optional[List[Player]] = None
    reserve_players: Optional[List[Player]] = None
    owner_id: Optional[str] = None
    associated_group_id: Optional[str] = None
    location: Optional[dict] = None
    status_list: bool = False
    price: Optional[float] = None
    price_type: Optional[Literal["per_person", "total_split"]] = None
